### PR TITLE
docs: add documentation for Image Factory

### LINKED ISSUE #7920: docs: Image Factory documentation
Describe how to use Image Factory to generate Talos boot assets.

### PR DESCRIPTION
**Pull request by smira**
Document Image Factory in general, and also provide specific examples for boot assets.

Secure Boot section is not covered, as we don't have Secure Boot support (yet) in the Image Factory.

Fixes #7920

### CODE REVIEW

File: website/content/v1.5/learn-more/image-factory.md [hunk 1]
```diff
@@ -0,0 +1,173 @@
+---
+title: "Image Factory"
+weight: 55
+description: "Image Factory generates customized Talos Linux images based on configured schematics."
+---
+
+The Image Factory provides a way to download Talos Linux artifacts.
```
Comment (frezbo) [inline website/content/v1.5/learn-more/image-factory.md:7]:
```suggestion
Image Factory provides a way to download Talos Linux artifacts.
```

having `The` feels weird at least for me

Comment (smira) [reply website/content/v1.5/learn-more/image-factory.md:7]: it's copy paste from https://github.com/siderolabs/image-factory#readme, and that's what Andrew put, so I trust him: https://github.com/siderolabs/image-factory/commit/91bbcd2c8217660a41078071c94ad742c58c3cfb#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5

**Review by frezbo — APPROVED** [review]

File: website/content/v1.5/learn-more/image-factory.md [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+    officialExtensions: # optional
+      - siderolabs/gvisor
+      - siderolabs/amd-ucode
+```
+
+The "vanilla" schematic is (it has an ID `376567988ad370138ad8b2698212367b8edcb69b5fd68c80be1f2ec7d603b4ba`):
+
+```yaml
+customization:
+```
```
Comment (TimJones) [inline website/content/v1.5/learn-more/image-factory.md:57]:
```suggestion
The "vanilla" schematic is:

```yaml
customization:
```
and has an ID of `376567988ad370138ad8b2698212367b8edcb69b5fd68c80be1f2ec7d603b4ba`.

```

File: website/content/v1.5/learn-more/image-factory.md [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+
+Models are Talos Linux images with customizations applied.
+The inputs to generate a model are:
+
+* schematic ID
+* Talos Linux version
+* model type (e.g. ISO, UKI, etc.)
+* architecture (e.g. amd64, arm64)
+* various model type specific options (e.g. disk image format, disk image size, etc.)
+
```
Comment (TimJones) [inline website/content/v1.5/learn-more/image-factory.md:77]: I feel that this should be a new section.

```suggestion

## Front ends

```

File: website/content/v1.5/learn-more/image-factory.md [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+Image Factory can be deployed on-premises to provide in-house asset generation.
+
+Image Factory requires following components:
+
+* an OCI registry to store schematics (private)
+* an OCI registry to store cached assets (private)
+* an OCI registry to store `installer` images (should allow public read-only access)
+* a container image signing key: ECDSA P-256 private key in PEM format
+
+Image Factory is configured using command like flags, use `--help` to see a list of available flags.
```
Comment (TimJones) [inline website/content/v1.5/learn-more/image-factory.md:166]:
```suggestion
Image Factory is configured using command line flags, use `--help` to see a list of available flags.
```

File: website/content/v1.5/talos-guides/install/boot-assets.md [hunk 5]
```diff
@@ -8,8 +8,132 @@ can be customized further for a specific use case:
 
 * adding [system extensions]({{< relref "../configuration/system-extensions" >}})
 * updating [kernel command line arguments]({{< relref "../../reference/kernel" >}})
+* using custom `META` contents, e.g. for [metal network configuration]({{< relref "../../advanced/metal-network-configuration" >}})
 * generating [SecureBoot]({{< relref "../install/bare-metal-platforms/secureboot" >}}) images signed with a custom key
 
+There are two ways to generate Talos boot assets:
+
+* using [Image Factory]({{< relref "#image-factory" >}}) service (recommended)
+* manually using [imager]({{< relref "#image-factory" >}}) container image (advanced)
```
Comment (TimJones) [inline website/content/v1.5/talos-guides/install/boot-assets.md:17]: This should be a reference to `imager` docs, right?

**Review by TimJones — APPROVED** [review]

File: website/content/v1.5/talos-guides/install/boot-assets.md (hunk 5 above)
Comment (smira) [reply website/content/v1.5/talos-guides/install/boot-assets.md:17]: yes, thanks

File: website/content/v1.5/learn-more/image-factory.md (hunk 4 above)
Comment (smira) [reply website/content/v1.5/learn-more/image-factory.md]: thanks, fixed

File: website/content/v1.5/learn-more/image-factory.md (hunk 2 above)
Comment (smira) [reply website/content/v1.5/learn-more/image-factory.md]: thanks, updated

**Pull request merged**